docs/README.md

```diff
@@ -71,7 +71,7 @@ The User Guide describes how to configure Triton, organize and configure your mo
 * Using the Client API [[Overview](README.md#client-libraries-and-examples) || [Details](https://github.com/triton-inference-server/client)]
 * Analyzing Performance [[Overview](README.md#performance-analysis)]
 * Deploying on edge (Jetson) [[Overview](README.md#jetson-and-jetpack)]
-
+* Debugging Guide [Details](./user_guide/debugging_guide.md)
```

Comment:
Looks good!

Nit: For consistency, it'd be good if this was something like "Debugging Failures [Overview]". If you do make this change, another nit unrelated to this that'd be great to update: "Buillding a Model Pipeline" above should be "Building a Model Pipeline".